Userland/Libraries/LibJS/Runtime/ArrayBufferPrototype.cpp

```diff
@@ -126,6 +127,51 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayBufferPrototype::slice)
     return new_array_buffer_object;
 }
 
+// 1.3.5 ArrayBuffer.prototype.resize ( newLength ), https://tc39.es/proposal-resizablearraybuffer/#sec-arraybuffer.prototype.resize
+JS_DEFINE_NATIVE_FUNCTION(ArrayBufferPrototype::resize)
+{
+    // 1. Let O be the this value.
+    auto* array_buffer_object = TRY(typed_this_value(global_object));
+
+    // 2. Perform ? RequireInternalSlot(O, [[ArrayBufferMaxByteLength]]).
+    if (!array_buffer_object->max_byte_length().has_value()) {
+        // 10.1.15 RequireInternalSlot ( O, internalSlot ), https://tc39.es/ecma262/#sec-requireinternalslot
+        // 1. If Type(O) is not Object, throw a TypeError exception.
+        // 2. If O does not have an internalSlot internal slot, throw a TypeError exception.
```

Comment:
I don't think we need the full definition of `RequireInternalSlot` here, we don't do that anywhere :)